nifi-nar-bundles/nifi-iceberg-bundle/nifi-iceberg-processors/src/main/java/org/apache/nifi/processors/iceberg/converter/IcebergRecordConverter.java

```diff
@@ -123,8 +135,12 @@ private static class IcebergSchemaVisitor extends SchemaWithPartnerVisitor<DataT
             // set NiFi schema field names (sourceFieldName) in the data converters
             for (DataConverter<?, ?> converter : converters) {
                 final Optional<String> mappedFieldName = recordType.getNameMapping(converter.getTargetFieldName());
-                final Optional<RecordField> recordField = recordSchema.getField(mappedFieldName.get());
-                converter.setSourceFieldName(recordField.get().getFieldName());
+                if (mappedFieldName.isEmpty()) {
+                    converter.setSourceFieldName(converter.getTargetFieldName());
+                } else {
+                    final Optional<RecordField> recordField = recordSchema.getField(mappedFieldName.get());
+                    converter.setSourceFieldName(recordField.get().getFieldName());
+                }
```

Comment:
I think we shouldn't set the `target field name` to the` source field name` because it is confusing. The `source name` represents the `NiFi record field name` and the `target name `represents the `iceberg record field name`. We should leave it null and set only the target.
```suggestion
                if (mappedFieldName.isPresent()) {
                    final Optional<RecordField> recordField = recordSchema.getField(mappedFieldName.get());
                    converter.setSourceFieldName(recordField.get().getFieldName());
                }
```

Reply:
I think the target field name here is necessary because the source field is missing right?

Reply:
Yes, you don't need to set it since you have it already in the targetFieldName attribute.

Reply:
This still needs to be addressed.